CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/value/expression/ExpressionParser.java

```diff
@@ -439,192 +440,75 @@ protected void print(final String prefix, final StringBuilder out) {
-                            contentLength = 0;
-                            state = ParseState.READING_WHITESPACE;
-                        }
-                        case '\\' -> state = ParseState.READING_DOUBLE_ESCAPED;
-                        case 0 -> {
-                            final String errorMessage = String.format(END_OF_QUOTED_STRING_ERROR, '"');
-                            if (!isHeadless) {
-                                Platform.runLater(() -> NotifyDisplayer.displayAlert(QUERY_ERROR,
-                                        MALFORMED_QUERY, errorMessage, Alert.AlertType.ERROR));
-                            }
-                            LOGGER.log(Level.WARNING, QUERY_ERROR_MSG, errorMessage);
-                            return null;
-                        }
-                        default -> content[contentLength++] = c;
+                    if (!parseReadingDoubleString(c, content, info)) {
+                        return null;
                     }
                 }
                 case READING_SINGLE_ESCAPED -> {
-                    if (c == 0) {
-                        final String errorMessage = "Found escaped character ' at end of expression.";
-                        if (!isHeadless) {
-                            Platform.runLater(() -> NotifyDisplayer.displayAlert(QUERY_ERROR,
-                                    MALFORMED_QUERY, errorMessage, Alert.AlertType.ERROR));
-                        }
-                        LOGGER.log(Level.WARNING, QUERY_ERROR_MSG, errorMessage);
+                    if (!parseReadingStringEscaped(c, content, info)) {
                         return null;
-                    } else {
-                        content[contentLength++] = c;
-                        state = ParseState.READING_SINGLE_STRING;
                     }
                 }
                 case READING_DOUBLE_ESCAPED -> {
-                    if (c == 0) {
-                        final String errorMessage = "Found escaped character \" at end of expression.";
-                        if (!isHeadless) {
-                            Platform.runLater(() -> NotifyDisplayer.displayAlert(QUERY_ERROR,
-                                    MALFORMED_QUERY, errorMessage, Alert.AlertType.ERROR));
-                        }
-                        LOGGER.log(Level.WARNING, QUERY_ERROR_MSG, errorMessage);
+                    if (!parseReadingDoubleEscaped(c, content, info)) {
                         return null;
-                    } else {
-                        content[contentLength++] = c;
-                        state = ParseState.READING_DOUBLE_STRING;
                     }
                 }
             }
         }
-
-        if (currentExpression != rootExpression) {
+        //Preventing sonar check, as we do want to check if the two variables are NOT the same object
+        if (info.currentExpression != info.rootExpression) {//NOSONAR
```

Comment:
I would rather mark this within sonar as accepted rather than adding the comment here (as should other smells pop up, the tag will nullify those as well). The comment you've added above though is still useful to retain (perhaps minus the sonar mention)